src/OrchardCoreContrib.PoExtractor.DotNet.CS/SingularStringExtractor.cs

```diff
@@ -41,13 +41,7 @@ public override bool TryExtract(SyntaxNode node, out LocalizableStringOccurence
                 var argument = accessor.ArgumentList.Arguments.FirstOrDefault();
                 if (argument != null && argument.Expression is LiteralExpressionSyntax literal && literal.IsKind(SyntaxKind.StringLiteralExpression))
                 {
-                    result = new LocalizableStringOccurence()
-                    {
-                        Text = literal.Token.ValueText,
-                        Context = MetadataProvider.GetContext(node),
-                        Location = MetadataProvider.GetLocation(node)
-                    };
-
+                    result = CreateLocalizedString(literal.Token.ValueText, null, node);
```

Comment:
Why? How this was working before?

Reply:
The bug was in the base CreateLocalizedString method. The private _metadataProvider was never set, but I assumed that was just a mistake and the protected MetadataProvider should be used instead.

The thing is, only PluralStringExtractor and LiquidStringExtractor used the base CreateLocalizedString method, so it may not have been noticed on a solution if it didn't use plural or Liquid. There are Liquid tests so I'm not sure why they didn't catch this. As part of this change I changed all the extractors to use the base method so there would be less code duplication.

I also noticed that there were tests for Liquid and DataAnnotations, but I didn't see any tests for IStringLocalizer/IHtmlLocalizer/IViewLocalizer calls, so those were the tests I added.

Reply:
I think [this](https://github.com/OrchardCoreContrib/OrchardCoreContrib.PoExtractor/commit/707c6f2c6b0faca9b97db28e86415019f98ee01a#diff-6ae9f09b59b80b25e622839e036baf3a611f5ccadefad9c01b13f41cb4a438b7) introduced the bug.